### PR TITLE
[#2234] Replace CSS animations for ticket layout so it can be based on number of retrieved tweets

### PR DESCRIPTION
**Pull request by mjschranz**
https://webmademovies.lighthouseapp.com/projects/65733/tickets/2234-twitter-ticker-animation-only-shows-4-or-5-tweets

### CODE REVIEW

File: templates/assets/plugins/twitter/popcorn.twitter.js [hunk 1]
```diff
@@ -200,7 +213,7 @@
           allTweets.push( results[ k ] );
         }
 
-        // Search API doesn't simply return count of tweets. It returns up to 100
+        // Search API doesn"t simply return count of tweets. It returns up to 100
```
Comment (cadecairos) [inline templates/assets/plugins/twitter/popcorn.twitter.js]: don't change the quote

File: templates/assets/plugins/twitter/popcorn.twitter.js [hunk 2]
```diff
@@ -271,6 +284,10 @@
       if ( options._container && options._target ) {
         options._target.removeChild( options._container );
       }
+
+      if ( options._interval ) {
```
Comment (cadecairos) [inline templates/assets/plugins/twitter/popcorn.twitter.js]: use more desciption in this var `options._tickerInterval` makes sense.

File: templates/assets/plugins/twitter/popcorn.twitter.js [hunk 3]
```diff
@@ -184,6 +185,19 @@
...
+            var elem;
+
+            options._tickerInterval = setInterval(function() {
+              elem = tweetsContainer.childNodes[ 0 ];
+              elem.style.marginTop = TRANSITION_MARGIN_TOP;
+              setTimeout(function() {
+                tweetsContainer.removeChild( elem );
+                tweetsContainer.appendChild( elem );
+                elem.style.marginTop = "";
+              }, 700 );
```
Comment (secretrobotron) [inline templates/assets/plugins/twitter/popcorn.twitter.js]: 700? Const it!

File: templates/assets/plugins/twitter/popcorn.twitter.js [hunk 4]
```diff
@@ -184,6 +185,19 @@
         // Set layout class for container
         if ( options.layout ) {
           options._container.classList.add( options.layout );
+          if ( options.layout === "ticker" ) {
+            var elem;
+
+            options._tickerInterval = setInterval(function() {
+              elem = tweetsContainer.childNodes[ 0 ];
```
Comment (secretrobotron) [inline templates/assets/plugins/twitter/popcorn.twitter.js]: Maybe you can use `firstChild` instead.

File: templates/assets/plugins/twitter/popcorn.twitter.js [hunk 5]
```diff
@@ -184,6 +185,19 @@
...
+              elem.style.marginTop = TRANSITION_MARGIN_TOP;
+              setTimeout(function() {
+                tweetsContainer.removeChild( elem );
+                tweetsContainer.appendChild( elem );
```
Comment (secretrobotron) [inline templates/assets/plugins/twitter/popcorn.twitter.js]: Nice hack :).

File: templates/assets/plugins/twitter/popcorn.twitter.css [hunk 6]
```diff
@@ -190,24 +102,16 @@
...
-  -o-animation-play-state: paused;
-  -ms-animation-play-state: paused;
-  animation-play-state: paused;
 }
 .popcorn-twitter.ticker ul > li {
   height: 55px;
   border: none;
   padding: 0;
   padding-left: 10px;
+  -webkit-transition: all 1s ease;
```
Comment (secretrobotron) [inline templates/assets/plugins/twitter/popcorn.twitter.css]: We can probably just get away with `margin-top`, right?

File: templates/assets/plugins/twitter/popcorn.twitter.js (hunk 5 above)
Comment (cadecairos) [inline templates/assets/plugins/twitter/popcorn.twitter.js]: oh man. doesn't removeChild return the elem it removes? thus: `tweetsContainer.appendChild( tweetsContainer.removeChild( elem ) );` would be valid. :P

**Pull request merged**